processing/src/main/java/org/apache/druid/query/filter/Filter.java

```diff
@@ -129,4 +131,52 @@ default boolean canVectorizeMatcher()
   {
     return false;
   }
+
+  /**
+   * Set of columns used by a filter
+   */
+  Set<String> getRequiredColumns();
+
+  /**
+   * "Manual" {@link FilterTuning} to allow explicit control of filter optimization behavior. This will likely be
+   * supplied by the {@link DimFilter} that is creating this {@link Filter}
+   */
+  @Nullable
+  FilterTuning getManualTuning();
+
+  /**
+   * This method allows a filter to automatically compute a {@link FilterTuning} based on information it can gather
```

Comment:
Similar comment to `computeTuning`, it seems like this might only be called by `shouldUseIndex`.

Reply:
Btw, what do you think about _not_ having `getManualTuning` and `computeTuning`, and instead having a static function in `Filters` that provides a useful default impl for `shouldUseIndex`? So most Filters would do something like:

```java
@Override
boolean shouldUseIndex(BitmapIndexSelector selector)
{
  return Filters.shouldUseIndex(this, selector, filterTuning);
}
```

It's not really more work or lines of code than overriding `getManualTuning`, which they'd have to do anyway with the current design. And it cuts 3 interface methods down to 1.

It would also be less spooky in terms of action-at-a-distance. It's nice to have methods defined in concrete classes rather than inherited from supertypes, since it makes it clearer to readers of the one class what is going on (without having to flip around as much between concrete impl and supertype).

Reply:
This makes sense, I think the current structure is a an artifact of my original version of this where the work was done by `QueryableIndexStorageAdapter`, will refactor

Reply:
done